Description: add fullscreen key
 update fullscreen function, to make go fullscreen and back, with the same key
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2020-04-27
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/main.cpp
+++ b/main.cpp
@@ -136,6 +136,7 @@
 				{
 				case SDLK_ESCAPE:done = true;break;
 				case SDLK_q: done = true; break;
+				case SDLK_f: win.ToggleFullscreen(); break;
 				default:
 					break;
 				}
--- a/win32.cpp
+++ b/win32.cpp
@@ -13,6 +13,13 @@
 {
 }
 
+void Window::ToggleFullscreen() {
+	int FullscreenFlag = SDL_WINDOW_FULLSCREEN_DESKTOP;
+	int IsFullscreen = SDL_GetWindowFlags(sdlWindow) & FullscreenFlag;
+	SDL_SetWindowFullscreen(sdlWindow, IsFullscreen ? 0 : FullscreenFlag);
+	SDL_ShowCursor(IsFullscreen);
+}
+
 bool Window::create(bool bflag)
 {
 	SDL_Init(SDL_INIT_VIDEO);
--- a/win32.hpp
+++ b/win32.hpp
@@ -12,6 +12,7 @@
 SDL_GLContext sdlGlcontext;
 bool create(bool bflag);
 void shut();
+void ToggleFullscreen();
 //		HDC getDC();
 
 private:
